pkg/operation/botanist/dnsrecord.go

```diff
@@ -206,3 +210,16 @@ func (b *Botanist) deployOrRestoreDNSRecord(ctx context.Context, dnsRecord compo
 	}
 	return dnsRecord.Deploy(ctx)
 }
+
+// CleanupOrphanedDNSRecordSecrets cleans up secrets related to DNSRecords which may be orphaned after introducing the 'dnsrecord-' prefix
+func (b *Botanist) CleanupOrphanedDNSRecordSecrets(ctx context.Context) error {
+	err := client.IgnoreNotFound(b.K8sSeedClient.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: b.Shoot.GetInfo().Name + "-" + DNSInternalName, Namespace: b.Shoot.SeedNamespace}}))
+	if err != nil {
+		return err
+	}
```

Comment:
Nit: Slightly more idiomatic / popular in our codebase:
```suggestion
	if err := b.K8sSeedClient.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: b.Shoot.GetInfo().Name + "-" + DNSInternalName, Namespace: b.Shoot.SeedNamespace}}); client.IgnoreNotFound(err) != nil {
		return err
	}
```

Reply:
done, thanks!